src/libraries/System.Net.WebHeaderCollection/src/System/Net/WebHeaderCollection.cs

```diff
@@ -413,35 +363,10 @@ protected void AddWithoutValidate(string headerName, string? headerValue)
         {
             headerName = HttpValidationHelpers.CheckBadHeaderNameChars(headerName);
             headerValue = HttpValidationHelpers.CheckBadHeaderValueChars(headerValue);
-            if (_type == WebHeaderCollectionType.WebResponse)
-            {
-                if (headerValue != null && headerValue.Length > ushort.MaxValue)
-                {
-                    throw new ArgumentOutOfRangeException(nameof(headerValue), headerValue, SR.Format(CultureInfo.InvariantCulture, SR.net_headers_toolong, ushort.MaxValue));
-                }
-            }
             InvalidateCachedArrays();
             InnerCollection.Add(headerName, headerValue);
         }
 
-        internal void ThrowOnRestrictedHeader(string headerName)
```

Comment:
Why are we removing this? This has nothing to do with header value length. The exception says you should use specific header method instead the generic ones.

Reply:
Looks like the enum values in these conditions were never set, so this was effectively dead code.